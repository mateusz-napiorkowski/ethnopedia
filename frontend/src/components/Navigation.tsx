import React from "react"
import { Link, useLocation } from "react-router-dom"
import { ReactComponent as AngleRightIcon } from "../assets/icons/angleRight.svg"

const isId = (segment: string) => {
    return /^[0-9a-f]{24}$/i.test(segment)
}

const Navigation: React.FC = () => {
    const location = useLocation()
    let pathSegments = location.pathname.split("/").filter(Boolean)

    if (pathSegments.length > 0 && isId(pathSegments[pathSegments.length - 1])) {
        //pathSegments = pathSegments.slice(0, -1)
    }

    // const breadcrumbItems = pathSegments.map((segment, index, array) => {
    //     const isLast = index === array.length - 1
    //     const to = `/${array.slice(0, index + 1).join("/")}`

    //     const decodedSegment = decodeURIComponent(segment)
    //     const translatedSegment = translations[decodedSegment.toLowerCase()] || decodedSegment
    //     const displaySegment = capitalizeFirstLetter(translatedSegment)

    //     return (
    //         <li key={index} className="flex items-center space-x-1 md:space-x-2">
    //             {isLast ? (
    //                 <span className="text-lg font-medium text-gray-500">{displaySegment}</span>
    //             ) : (
    //                 <>
    //                     <Link to={to}
    //                           className="inline-flex items-center text-lg font-medium text-gray-800 hover:text-blue-600">
    //                         {displaySegment}
    //                     </Link>
    //                     <AngleRightIcon />
    //                 </>
    //             )}
    //         </li>
    //     )
    // })

    return <nav className="flex">
        <ol className="inline-flex items-center space-x-1 md:space-x-2">
            <li className="inline-flex items-center text-lg">
                <Link to="/"
                      className="inline-flex items-center font-medium text-gray-800 hover:text-blue-600">
                    Powrót do listy kolekcji
                </Link>
            </li>
            <span className="self-center">
                <AngleRightIcon />
            </span>
            <li className="inline-flex items-center text-lg">
                <Link to={`/collections/${pathSegments[1]}/artworks`}
                      className="inline-flex items-center font-medium text-gray-800 hover:text-blue-600">
                    {pathSegments[pathSegments.length-1] === "artworks" ? pathSegments[1] : "Powrót do listy utworów"}
                </Link>
            </li>
            {/* {breadcrumbItems} */}
        </ol>
    </nav>
}

export default Navigation
