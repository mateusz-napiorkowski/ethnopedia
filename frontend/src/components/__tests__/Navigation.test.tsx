import '@testing-library/jest-dom';
import { render } from '@testing-library/react'
import Navigation from "../Navigation"
import { MemoryRouter, Routes, Route } from "react-router-dom";

interface locationState {
    categories: Array<{
        name: string,
        value: string
    }>
}

const exampleCollectionId = "67f84d80d2ac8e9a1e67cca4"
const renderComponent = (url: string, title: string | undefined = undefined, locationState: locationState | undefined = undefined) => {
    return render(
        <MemoryRouter initialEntries={[{pathname: url, state: locationState}]}>
            <Routes>
                <Route path={url} element={<Navigation/>}/>
            </Routes>  
        </MemoryRouter>
    );
};

describe("Navigation tests", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    test("component renders correctly on artworks list page", async () => {        
        const url = `/collections/${exampleCollectionId}/artworks/`
        const { container } = renderComponent(url)
        
        expect(container).toMatchSnapshot()
    })

    test("component renders correctly on artwork page", async () => {        
        const url = `/collections/${exampleCollectionId}/artworks/674386b32a2908778c0ad471`
        const { container } = renderComponent(url, 'Example artwork title')
        
        expect(container).toMatchSnapshot()
    })

    test("component renders correctly on create artwork page", async () => {        
        const url = `/collections/${exampleCollectionId}/create-artwork`
        const { container } = renderComponent(url)
        
        expect(container).toMatchSnapshot()
    })

    test("component renders correctly on edit artwork page", async () => {        
        const url = `/collections/${exampleCollectionId}/artworks/674386b32a2908778c0ad471/edit-artwork`
        const state = {categories: [{name: "TytuÅ‚", value: 'Example artwork title'}]}
        const { container } = renderComponent(url, undefined, state)
        
        expect(container).toMatchSnapshot()
    })

    test("component renders correctly on export data page", async () => {        
        const url = `/collections/${exampleCollectionId}/export-data`
        const { container } = renderComponent(url)
        
        expect(container).toMatchSnapshot()
    })
})