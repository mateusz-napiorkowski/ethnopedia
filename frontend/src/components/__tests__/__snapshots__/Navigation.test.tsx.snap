// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation tests component renders correctly on artwork page 1`] = `
<div>
  <nav
    class="flex"
  >
    <ol
      class="inline-flex items-center space-x-1 md:space-x-2"
    >
      <li
        class="inline-flex items-center text-lg"
      >
        <a
          class="inline-flex items-center font-medium text-gray-600 dark:text-gray-400 hover:text-blue-600 dark:hover:text-blue-600"
          href="/"
        >
          Strona główna
        </a>
      </li>
      <span
        class="self-center"
      >
        <svg
          class="w-3 h-3 text-gray-600 dark:text-gray-400"
        />
      </span>
      <li
        class="inline-flex items-center text-lg"
      >
        <a
          class="inline-flex items-center font-medium text-gray-600 dark:text-gray-400 hover:text-blue-600 dark:hover:text-blue-600"
          href="/collections/67f84d80d2ac8e9a1e67cca4/artworks"
        >
          Kolekcja
        </a>
      </li>
      <span
        class="self-center"
      >
        <svg
          class="w-3 h-3 text-gray-600 dark:text-gray-400"
        />
      </span>
      <li
        class="inline-flex items-center text-lg"
      >
        <a
          class="inline-flex items-center font-medium text-gray-600 dark:text-gray-400 hover:text-blue-600 dark:hover:text-blue-600"
          href="/collections/67f84d80d2ac8e9a1e67cca4/artworks/674386b32a2908778c0ad471"
        >
          Rekord
        </a>
      </li>
    </ol>
  </nav>
</div>
`;

exports[`Navigation tests component renders correctly on artworks list page 1`] = `
<div>
  <nav
    class="flex"
  >
    <ol
      class="inline-flex items-center space-x-1 md:space-x-2"
    >
      <li
        class="inline-flex items-center text-lg"
      >
        <a
          class="inline-flex items-center font-medium text-gray-600 dark:text-gray-400 hover:text-blue-600 dark:hover:text-blue-600"
          href="/"
        >
          Strona główna
        </a>
      </li>
      <span
        class="self-center"
      >
        <svg
          class="w-3 h-3 text-gray-600 dark:text-gray-400"
        />
      </span>
      <li
        class="inline-flex items-center text-lg"
      >
        <a
          class="inline-flex items-center font-medium text-gray-600 dark:text-gray-400 hover:text-blue-600 dark:hover:text-blue-600"
          href="/collections/67f84d80d2ac8e9a1e67cca4/artworks"
        >
          Kolekcja
        </a>
      </li>
    </ol>
  </nav>
</div>
`;

exports[`Navigation tests component renders correctly on create artwork page 1`] = `
<div>
  <nav
    class="flex"
  >
    <ol
      class="inline-flex items-center space-x-1 md:space-x-2"
    >
      <li
        class="inline-flex items-center text-lg"
      >
        <a
          class="inline-flex items-center font-medium text-gray-600 dark:text-gray-400 hover:text-blue-600 dark:hover:text-blue-600"
          href="/"
        >
          Strona główna
        </a>
      </li>
      <span
        class="self-center"
      >
        <svg
          class="w-3 h-3 text-gray-600 dark:text-gray-400"
        />
      </span>
      <li
        class="inline-flex items-center text-lg"
      >
        <a
          class="inline-flex items-center font-medium text-gray-600 dark:text-gray-400 hover:text-blue-600 dark:hover:text-blue-600"
          href="/collections/67f84d80d2ac8e9a1e67cca4/artworks"
        >
          Kolekcja
        </a>
      </li>
      <span
        class="self-center"
      >
        <svg
          class="w-3 h-3 text-gray-600 dark:text-gray-400"
        />
      </span>
      <li
        class="inline-flex items-center text-lg"
      >
        <a
          class="inline-flex items-center font-medium text-gray-600 dark:text-gray-400 hover:text-blue-600 dark:hover:text-blue-600"
          href="/collections/67f84d80d2ac8e9a1e67cca4/create-artwork"
        >
          Dodaj nowy rekord
        </a>
      </li>
    </ol>
  </nav>
</div>
`;

exports[`Navigation tests component renders correctly on edit artwork page 1`] = `
<div>
  <nav
    class="flex"
  >
    <ol
      class="inline-flex items-center space-x-1 md:space-x-2"
    >
      <li
        class="inline-flex items-center text-lg"
      >
        <a
          class="inline-flex items-center font-medium text-gray-600 dark:text-gray-400 hover:text-blue-600 dark:hover:text-blue-600"
          href="/"
        >
          Strona główna
        </a>
      </li>
      <span
        class="self-center"
      >
        <svg
          class="w-3 h-3 text-gray-600 dark:text-gray-400"
        />
      </span>
      <li
        class="inline-flex items-center text-lg"
      >
        <a
          class="inline-flex items-center font-medium text-gray-600 dark:text-gray-400 hover:text-blue-600 dark:hover:text-blue-600"
          href="/collections/67f84d80d2ac8e9a1e67cca4/artworks"
        >
          Kolekcja
        </a>
      </li>
      <span
        class="self-center"
      >
        <svg
          class="w-3 h-3 text-gray-600 dark:text-gray-400"
        />
      </span>
      <li
        class="inline-flex items-center text-lg"
      >
        <a
          class="inline-flex items-center font-medium text-gray-600 dark:text-gray-400 hover:text-blue-600 dark:hover:text-blue-600"
          href="/collections/67f84d80d2ac8e9a1e67cca4/artworks/674386b32a2908778c0ad471"
        >
          Rekord
        </a>
      </li>
      <span
        class="self-center"
      >
        <svg
          class="w-3 h-3 text-gray-600 dark:text-gray-400"
        />
      </span>
      <li
        class="inline-flex items-center text-lg"
      >
        <a
          class="inline-flex items-center font-medium text-gray-600 dark:text-gray-400 hover:text-blue-600 dark:hover:text-blue-600"
          href="/collections/67f84d80d2ac8e9a1e67cca4/artworks/674386b32a2908778c0ad471/edit-artwork"
        >
          Edycja
        </a>
      </li>
    </ol>
  </nav>
</div>
`;

exports[`Navigation tests component renders correctly on export data page 1`] = `
<div>
  <nav
    class="flex"
  >
    <ol
      class="inline-flex items-center space-x-1 md:space-x-2"
    >
      <li
        class="inline-flex items-center text-lg"
      >
        <a
          class="inline-flex items-center font-medium text-gray-600 dark:text-gray-400 hover:text-blue-600 dark:hover:text-blue-600"
          href="/"
        >
          Strona główna
        </a>
      </li>
      <span
        class="self-center"
      >
        <svg
          class="w-3 h-3 text-gray-600 dark:text-gray-400"
        />
      </span>
      <li
        class="inline-flex items-center text-lg"
      >
        <a
          class="inline-flex items-center font-medium text-gray-600 dark:text-gray-400 hover:text-blue-600 dark:hover:text-blue-600"
          href="/collections/67f84d80d2ac8e9a1e67cca4/artworks"
        >
          Kolekcja
        </a>
      </li>
      <span
        class="self-center"
      >
        <svg
          class="w-3 h-3 text-gray-600 dark:text-gray-400"
        />
      </span>
      <li
        class="inline-flex items-center text-lg"
      >
        <a
          class="inline-flex items-center font-medium text-gray-600 dark:text-gray-400 hover:text-blue-600 dark:hover:text-blue-600"
          href="/collections/67f84d80d2ac8e9a1e67cca4/export-data"
        >
          Eksportuj dane
        </a>
      </li>
    </ol>
  </nav>
</div>
`;
